lua/starfall/libs_sh/mesh.lua

```diff
@@ -762,6 +775,191 @@ if CLIENT then
+		end
+		mesh.End()
+		
+		meshd.started = false
+	end
+	
+	--- Sets the vertex color by RGBA values
+	-- @client
+	function mesh_library.writeColor(r, g, b, a)
+		local meshd = instance.data.mesh_dynamic
+		if not meshd.started then SF.Throw("Dyanic mesh not started.", 2) end
+		
+		checkluatype(r, TYPE_NUMBER)
+		checkluatype(g, TYPE_NUMBER)
+		checkluatype(b, TYPE_NUMBER)
+		checkluatype(a, TYPE_NUMBER)
+		
+		meshd.modified_vertex = true
+		table.insert(meshd.instructions, {mesh.AdvanceVertex, {r, g, b, a}})
+	end
+	
+	--- Sets the vertex normal
+	-- @param normal Vector
+	-- @client
+	function mesh_library.writeNormal(norm)
+		local meshd = instance.data.mesh_dynamic
+		if not meshd.started then SF.Throw("Dyanic mesh not started.", 2) end
+		
+		meshd.modified_vertex = true
+		table.insert(meshd.instructions, {mesh.Normal, {vunwrap(norm)}})
+	end
+	
+	--- Sets the vertex position
+	-- @param position Vector
+	-- @client
+	function mesh_library.writePosition(pos)
+		local meshd = instance.data.mesh_dynamic
+		if not meshd.started then SF.Throw("Dyanic mesh not started.", 2) end
+		
+		meshd.modified_vertex = true
+		table.insert(meshd.instructions, {mesh.Position, {vunwrap(pos)}})
+	end
+	
+	--- Sets the vertex texture coordinates
+	-- @param stage Number, stage of the texture coordinate
+	-- @param u Number, u coordinate
+	-- @param v Number, v coordinate
+	-- @client
+	function mesh_library.writeUV(stage, u, v)
+		local meshd = instance.data.mesh_dynamic
+		if not meshd.started then SF.Throw("Dyanic mesh not started.", 2) end
+		
+		checkluatype(stage, TYPE_NUMBER)
+		checkluatype(u, TYPE_NUMBER)
+		checkluatype(v, TYPE_NUMBER)
+		
+		meshd.modified_vertex = true
+		table.insert(meshd.instructions, {mesh.TexCoord, {stage, u, v}})
+	end
+	
+	--- Sets the vertex user data
+	-- @param tangent_x
+	-- @param tangent_y
+	-- @param tangent_z
+	-- @param tangent_handedness
+	-- @client
+	function mesh_library.writeUserData(x, y, z, handedness)
+		local meshd = instance.data.mesh_dynamic
+		if not meshd.started then SF.Throw("Dyanic mesh not started.", 2) end
+		
+		checkluatype(x, TYPE_NUMBER)
+		checkluatype(y, TYPE_NUMBER)
+		checkluatype(z, TYPE_NUMBER)
+		checkluatype(handedness, TYPE_NUMBER)
+		
+		meshd.modified_vertex = true
+		table.insert(meshd.instructions, {mesh.UserData, {x, y, z, handedness}})
+	end
+	
+	--- Draws a quad using 4 vertices
+	-- @client
+	function mesh_library.writeQuad(v1, v2, v3, v4)
+		local meshd = instance.data.mesh_dynamic
+		if not meshd.started then SF.Throw("Dyanic mesh not started.", 2) end
+		
+		meshd.modified_vertex = true
+		meshd.true_vertices = meshd.true_vertices + 4
+		table.insert(meshd.instructions, {mesh.Quad, {vunwrap(v1), vunwrap(v2), vunwrap(v3), vunwrap(v4)}})
+	end
+	
+	--- Draws a quad using a position, normal and size
+	-- @param position Vector
+	-- @param normal Vector
+	-- @param w Number
+	-- @param h Number
+	-- @client
+	function mesh_library.writeQuadEasy(pos, norm, w, h)
```

Comment:
Use same name of param as in func, while it's not used ATM(at least that I know it) it might be in future